### PR TITLE
stop auto-submit when exam ends early

### PR DESCRIPTION
**Pull request by doombot-exe**

### CODE REVIEW

**Review by PanopticaRising — APPROVED** [review]

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -3473,20 +3473,23 @@ class CourseController {
...
-            // }
+            }            
 
             return { problemScores: problemScoresReturn, bestVersionScore: bestVersionScoreReturn, bestOverallVersion: bestOverallVersionReturn};
         });
     };
+
+    endAssessmentEarly(studentTopicAssessmentInfo: StudentTopicAssessmentInfo, wasAutoSubmitted: boolean): void {
+        studentTopicAssessmentInfo.isClosed = true;
+        studentTopicAssessmentInfo.save();
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:3484]: Need to await

File: src/features/courses/course-route.ts [hunk 2]
```diff
@@ -343,12 +343,8 @@ router.get('/assessment/topic/end/:id',
             throw new ForbiddenError('You cannot end an exam without making at least one attempt.');
         }
 
-        // if (version.isClean === false) {
-        //     // throw error? the student has made changes to their answers, but they have not submitted those changes
-        // }
+        courseController.endAssessmentEarly(version, true);
```
Comment (red-tt-bear) [inline src/features/courses/course-route.ts:346]: needs await once asyned

Comment (red-tt-bear) [reply src/features/courses/course-route.ts:346]: (and in the other spot as well)

**Review by red-tt-bear — CHANGES_REQUESTED** [review]
async await

**Review by red-tt-bear — APPROVED** [review]

**Pull request merged**